src/mangosd/RASocket.cpp

```diff
@@ -210,5 +210,6 @@ bool RASocket::HandleInput()
 
 void RASocket::Send(const std::string& message)
 {
-    Write(message.c_str(), message.length());
+    auto self(shared_from_this());
+    Write(message.c_str(), message.length(), [self](const boost::system::error_code& error, std::size_t read) {});
```

Comment:
This looks dangerous, with ```message``` being a reference. You need to keep that alive during the async call and there's no guarantee the caller is doing that!

Reply:
Yeah that went under my radar and this is precisely the kind of thing why its in the PR. Ty, will absolutely make that a shared ptr like the rest.